azurerm/resource_arm_connection_monitor.go

```diff
@@ -0,0 +1,336 @@
+		existing, err := client.Get(ctx, resourceGroup, watcherName, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Error checking for presence of existing Connection Monitor %q (Watcher %q / Resource Group %q): %s", name, watcherName, resourceGroup, err)
+			}
+		}
+
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_connection_monitor", *existing.ID)
+		}
+	}
+
+	tags := d.Get("tags").(map[string]interface{})
+
+	properties := network.ConnectionMonitor{
+		Location: utils.String(location),
+		Tags:     expandTags(tags),
+		ConnectionMonitorParameters: &network.ConnectionMonitorParameters{
+			Source:                      source,
+			Destination:                 dest,
+			AutoStart:                   utils.Bool(autoStart),
+			MonitoringIntervalInSeconds: utils.Int32(intervalInSeconds),
+		},
+	}
+
+	future, err := client.CreateOrUpdate(ctx, resourceGroup, watcherName, name, properties)
+	if err != nil {
+		return fmt.Errorf("Error creating Connection Monitor %q (Watcher %q / Resource Group %q): %+v", name, watcherName, resourceGroup, err)
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error waiting for completion of Connection Monitor %q (Watcher %q / Resource Group %q): %+v", name, watcherName, resourceGroup, err)
+	}
+
+	resp, err := client.Get(ctx, resourceGroup, watcherName, name)
+	if err != nil {
+		return fmt.Errorf("Error retrieving Connection Monitor %q (Watcher %q / Resource Group %q): %+v", name, watcherName, resourceGroup, err)
+	}
+	if resp.ID == nil {
+		return fmt.Errorf("Cannot read Connection Monitor %q (Watcher %q / Resource Group %q) ID", name, watcherName, resourceGroup)
+	}
+
+	d.SetId(*resp.ID)
+
+	return resourceArmConnectionMonitorRead(d, meta)
+}
+
+func resourceArmConnectionMonitorRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).connectionMonitorsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	watcherName := id.Path["networkWatchers"]
+	name := id.Path["connectionMonitors"]
+
+	resp, err := client.Get(ctx, resourceGroup, watcherName, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error reading Connection Monitor %q (Watcher %q / Resource Group %q) %+v", name, watcherName, resourceGroup, err)
+	}
+
+	d.Set("name", name)
+	d.Set("network_watcher_name", watcherName)
+	d.Set("resource_group_name", resourceGroup)
+	if location := resp.Location; location != nil {
+		d.Set("location", azureRMNormalizeLocation(*location))
+	}
+
+	if props := resp.ConnectionMonitorResultProperties; props != nil {
+		d.Set("auto_start", props.AutoStart)
+		d.Set("interval_in_seconds", int(*props.MonitoringIntervalInSeconds))
```

Comment:
If `d.Set("interval_in_seconds", props.MonitoringIntervalInSeconds)` doesn't work and the cast is required, we will need to nil check before doing the dereference

Reply:
I recall fighting with this but I am unable to reproduce any reason for the cast today, so I will remove it.